### PR TITLE
fix(inspector): call call_end/create_end when inspector shortcircuits calls

### PR DESCRIPTION
**Pull request by Evalir**
If the Inspector shortcircuits the call of `call/create`, functions for call end `call_end/create_end` were not called, these calls were expected by the foundry and it broke a few things.

### CODE REVIEW

File: crates/revm/src/evm_impl.rs [hunk 1]
```diff
@@ -415,16 +415,55 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
...
     }
 
+    fn create_end(
+        &mut self,
+        inputs: &CreateInputs,
+        ret: InstructionResult,
+        address: Option<Address>,
+        gas: Gas,
+        out: Bytes,
+    ) -> (InstructionResult, Option<Address>, Gas, Bytes) {
```
Comment (rakita) [inline crates/revm/src/evm_impl.rs:425]: Can you revert back to how it was, as this function (and others `create_precheck `,`call_..`) just wrap around inspector calls and hides just very small functionality and is called only from one place, it is not worth imo.

Comment (rakita) [reply crates/revm/src/evm_impl.rs]: Not exactly what I asked, will make a change to push it over the line.

**Review by rakita — APPROVED** [review]
@Evalir can you double check the PR?

**Comment by Evalir** [comment]
@rakita yep double checking rn

**Review by Evalir — COMMENTED** [review]
Alright @rakita — triple checked, confirmed it works as expected. Thank you!

**Pull request merged**